### PR TITLE
Implement basic support for POWER10 in the compiler

### PR DESCRIPTION
**Pull request by aviansie-ben**
This PR adds basic support for detecting POWER10 systems to the OpenJ9 JIT. At this time, POWER10 support in the compiler is gated behind an environment variable to prevent accidental future use of this functionality by users who use older versions of OpenJ9 on future POWER10 systems, since the current support is untested. Currently, the only exploitation which has been added is the use of the `brh` and `brd` instructions for the recognized `{Short,Long}.reverseBytes` methods, but more will be added in the near future.

### CODE REVIEW

**Comment by zl-wang** [comment]
Looks good to me.

File: runtime/compiler/env/ProcessorDetection.cpp [hunk 1]
```diff
@@ -269,6 +269,20 @@ TR_Processor mapJ9Processor(J9ProcessorArchitecture j9processor)
...
          tp = TR_PPCp9;
          break;
+      case PROCESSOR_PPC_P10:
+         // P10 support is not yet well-tested, so it's currently gated behind an environment
+         // variable to prevent it from being used by accident by users who use old versions of
+         // OpenJ9 once P10 chips become available.
+         if (jitConfig)
+            {
+            static bool enableP10 = feGetEnv("TR_EnableExperimentalPower10Support");
+            tp = enableP10 ? TR_PPCp10 : TR_PPCp9;
```
Comment (zl-wang) [inline runtime/compiler/env/ProcessorDetection.cpp:279]: older version of OpenJ9 cannot recognize P10 such that it is only an Unknown processor (i.e. DefaultProcessor). This code on P10Processor never kicks in anyway such that It is unnecessary.

Comment (aviansie-ben) [reply runtime/compiler/env/ProcessorDetection.cpp:279]: Sorry, I should clarify: by "older versions", I'm referring to versions that will be released between when this PR is merged and when the P10 support is fully tested and verified. If a user were to try and use one of those builds on a newer machine with a POWER10 chip, OpenJ9 would recognize it and enable the new features, despite them having not yet undergone testing.

Comment (zl-wang) [reply runtime/compiler/env/ProcessorDetection.cpp:279]: providing a testing knob might make more senses, before it is enabled by default.

**Comment by aviansie-ben** [comment]
Rebased and removed WIP now that eclipse/omr#5184 has merged.

**Comment by pshipton** [comment]
https://github.com/eclipse/omr/pull/5184 needs to promote first. The next omr acceptance build is 4pm today. Do the OMR changes build without this change?

**Comment by aviansie-ben** [comment]
Ah, right, forgot that it needs to promote first. No, this will not build correctly without the corresponding OMR change.

**Comment by pshipton** [comment]
> this will not build correctly without the corresponding OMR change.

My question was actually the reverse. Will the OMR change build successfully without this one? If not, the OMR acceptance build will fail and we'll need to take other action.

**Comment by aviansie-ben** [comment]
Oops, misread your question. My mistake :sweat_smile:. Yes, the OMR change should build fine without this one.

**Comment by aviansie-ben** [comment]
OMR change promoted last night, so this is now _actually_ ready to go.

**Comment by harryyu1994** [comment]
Hi, I'm wondering when this will be merged.

**Comment by zl-wang** [comment]
Jenkins test sanity plinux,aix

**Comment by zl-wang** [comment]
Jenkins test sanity plinux,aix jdk8,jdk11

**Comment by zl-wang** [comment]
the lone AIX failure is caused by Jenkins infrastructure restart. (AIX java8/Linux Java8 and 11 all passed) I am not going to relaunch the whole tests.

**Pull request merged**